From 2facef4b0bee8ac0ccd433ec9905d999cf6c65e3 Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Fri, 3 Feb 2012 19:35:36 +0000
Subject: Don't use _POSIX_PATH_MAX as a fallback PATH_MAX

_POSIX_PATH_MAX is required to be defined for POSIX systems as
256, which is too small for a fallback PATH_MAX (some systems, such as
GNU/Hurd, do not have an inherent limit on path length).

Bug-Debian: http://bugs.debian.org/656869
Bug: https://rt.perl.org/rt3/Public/Bug/Display.html?id=109262
Origin: https://rt.perl.org/rt3/Public/Bug/Display.html?id=109262

Patch-Name: fixes/path_max_fallback.diff
---
 perl.h |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/perl.h b/perl.h
index facdacf..53f0651 100644
--- a/perl.h
+++ b/perl.h
@@ -2778,11 +2778,7 @@ freeing any remaining Perl interpreters.
 #      define MAXPATHLEN (PATH_MAX+1)
 #    endif
 #  else
-#    ifdef _POSIX_PATH_MAX
-#       define MAXPATHLEN _POSIX_PATH_MAX
-#    else
-#       define MAXPATHLEN 1024	/* Err on the large side. */
-#    endif
+#    define MAXPATHLEN 1024	/* Err on the large side. */
 #  endif
 #endif
 
